lighthouse-core/audits/byte-efficiency/unused-images.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ /**
+  * @fileoverview Checks to see if images are displayed only outside of the viewport.
+  */
+'use strict';
+
+const Audit = require('./byte-efficiency-audit');
+const URL = require('../../lib/url-shim');
+
+const ALLOWABLE_OFFSCREEN_X = 100;
+const ALLOWABLE_OFFSCREEN_Y = 200;
+
+const IGNORE_THRESHOLD_IN_BYTES = 2048;
+
+class UnusedImages extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Images',
+      name: 'unused-images',
+      description: 'Unused images',
+      helpText: 'Images that are not above the fold should be lazy loaded on interaction',
+      requiredArtifacts: ['ImageUsage', 'ViewportDimensions', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {{top: number, right: number, bottom: number, left: number}} imageRect
+   * @param {{scrollWidth: number, scrollHeight: number}} viewportDimensions
+   * @return {number}
+   */
+  static computeVisiblePixels(imageRect, viewportDimensions) {
+    const scrollWidth = viewportDimensions.scrollWidth;
```

Comment:
Not clientWidth/Height of the viewport? scrollWidth/Height includes off screen content.